File: backend/webapicsharp/Modelos/EcoPunto.cs
using System.ComponentModel.DataAnnotations;

namespace webapicsharp.Modelos
{
    public class EcoPunto
    {
        public int Id { get; set; }
        public string? Horario { get; set; }
        public string? Nombre { get; set; }
        public Trabajador? Trabajador { get; set; }
        public List<Material>? MaterialesAceptados { get; set; } = new List<Material>();
        public Ubicacion? Ubicacion { get; set; }
    }

    public class ActualizarEcoPuntoDto
    {
        public string? CodigoDeEmpleado { get; set; }

        public string? Latitud { get; set; }

        public string? Longitud { get; set; }

        public string? Direccion { get; set; }

        public string? Horario { get; set; }
        public string? Nombre { get; set; }
        public List<Material> Materiales { get; set; } = new();
    }
}

File: frontend/src/interfaces/ecopunto.ts
export interface Ecopunto {
    id: number
    nombre: string
    horario: string
    trabajador: Trabajador
    materialesAceptados: MaterialesAceptado[]
    ubicacion: Ubicacion
}

interface Trabajador {
    horario: string
    codigoDeEmpleado: string
    id: number
    nombre: string
    cedula: string
    correo: string
    direccion: string
    telefono: string
}

export interface MaterialesAceptado {
    id: number
    nombre: string
    peso: number
}

interface Ubicacion {
    latitud: number | string
    longitud: number | string
    direccion: string
}

export interface EcopuntoRequestGet {
    limite: number
}

export interface formDataInsert {
    nombre: string
    direccion: string
    latitud: number | string
    longitud: number | string
    materiales: string[]
    codigoDeEmpleado: string
}

export interface formDataUpdateForm extends formDataInsert {
    id: number
}

export interface formDataUpdateRequest {
    id: number
    nombre: string
    direccion: string
    latitud: string
    longitud: string
    materiales: MaterialesAceptado[]
    horario: string
    codigoDeEmpleado: string
}

export interface searchEcopuntoByIdResponse {
    mensaje: string
    ecoPunto: Ecopunto
}